cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/distributed/AbstractDistributedProgramRunner.java

```diff
@@ -229,6 +229,8 @@ public TwillController launch(TwillApplication twillApplication, Iterable<String
             // TokenSecureStoreUpdater.update() ignores parameters
             twillPreparer.addSecureStore(secureStoreUpdater.update(null, null));
           }
+
+          System.setProperty("user.name", cConf.get(Constants.CFG_HDFS_USER));
```

Comment:
Add a TODO to get this removed when the issue in Twill get fixed. This is because System properties are global and it can affect other operations.